ch5-catdir/utils.cpp

```diff
@@ -0,0 +1,58 @@
+#include "utils.h"
+
+#include <cctype>
+
+#include <sys/stat.h>
+
+namespace
+{
+
+auto typeChar(mode_t mode)
+{
+    if (S_ISREG(mode))  { return '-'; }
+    if (S_ISDIR(mode))  { return 'd'; }
+    if (S_ISLNK(mode))  { return 'l'; }
+    if (S_ISCHR(mode))  { return 'c'; }
+    if (S_ISBLK(mode))  { return 'b'; }
+    if (S_ISFIFO(mode)) { return 'p'; }
+    if (S_ISSOCK(mode)) { return 's'; }
+    return '?';
+}
+
+void setAdditionalFlag(char& target, char flag)
```

Comment:
Don't you prefer pure functions? Instead of changing the argument, return the result?

Reply:
I wrote it this way because the function has to read the current value of `target` to determine whether to use `flag` or `toupper(flag)`, and I don't want to specify the target twice.

Reply:
I agree this is a matter of taste